sdk/storage/Azure.Storage.Queues/src/QueueClientOptions.cs

```diff
@@ -113,6 +113,23 @@ public QueueClientOptions(ServiceVersion version = LatestVersion)
         /// </summary>
         public QueueMessageEncoding MessageEncoding { get; set; } = QueueMessageEncoding.None;
 
+        /// <summary>
+        /// Optional. Performs the tasks needed when an invalid message is received or peaked from the queue.
+        ///
+        /// <para>Invalid message can be received or peaked when <see cref="QueueClient"/> is expecting certain <see cref="QueueMessageEncoding"/>
+        /// but there's another producer that is not encoding messages in expected way. I.e. the queue contains messages with different encoding.</para>
+        ///
+        /// <para><see cref="InvalidMessageEventArgs"/> contains <see cref="QueueClient"/> that has received invalid message as well as the message
+        /// which can be either <see cref="QueueMessage"/> or <see cref="PeekedMessage"/> with raw body, i.e. no decoding will be attempted so that
+        /// body can be inspected as has been received from the queue.</para>
+        ///
+        /// <para>The <see cref="QueueClient"/> won't attempt to remove invalid message from the queue. Therefore such handling should be included into
+        /// the event handler itself.</para>
+        /// </summary>
+        public event SyncAsyncEventHandler<InvalidMessageEventArgs> OnInvalidMessage;
+
+        internal SyncAsyncEventHandler<InvalidMessageEventArgs> GetInvalidMessageHandlers() => OnInvalidMessage;
```

Comment:
Why do we need this method?

Reply:
I see we can't get the delegate list outside of the class. Should we freeze it in this method so future subscriptions don't affect existing clients?

Reply:
@tg-msft 's suggestion. This is to copy handlers from options to clients.
When I tried to just copy `event` I was hitting that:
![image](https://user-images.githubusercontent.com/61715331/106051037-c323cc80-609c-11eb-9055-73ca4113475c.png)

Reply:
Doesn't copying the value of the delegate mean future event subscriptions won't affect it?  (I'm thinking about the interlocked exchange in add/remove as my rationalization here.)

Reply:
They won't. I have added tests that proves that. QueueClient takes snapshot of subscriptions at creation.